### DISCUSSION TITLE
Seemingly redundant flush operation in ansicolor_sink

### DISCUSSION THREAD

**Discussion by ghost** [gabime/spdlog#3061] [Q&A] [answered]
It looks like `fflush` is already called after each message being logged. Why does the flush policy, if enabled, trigger an additional `fflush` operation?

```
template <typename ConsoleMutex>
SPDLOG_INLINE void ansicolor_sink<ConsoleMutex>::log(const details::log_msg &msg) {
    std::lock_guard<mutex_t> lock(mutex_);
    [...]
    {
        print_range_(formatted, 0, formatted.size());
    }
    fflush(target_file_);
}

template <typename ConsoleMutex>
SPDLOG_INLINE void ansicolor_sink<ConsoleMutex>::flush() {
    std::lock_guard<mutex_t> lock(mutex_);
    fflush(target_file_);
}
```

**Comment by tt4g** [comment, marked as answer]
I did not implement `ansicolor_sink`, but I found the maintainer's comment: https://github.com/gabime/spdlog/issues/2321#issuecomment-1079662200

> The stdout/stderr sinks flush on each log [regardless of the flush level](https://github.com/gabime/spdlog/blob/a49456f7f21fb802c1000fd79ef05911aa2101b7/include/spdlog/sinks/stdout_sinks-inl.h#L76). This is by design since normally users expect to see the screen output immediately, and also due to the fact that under windows the `WriteConsoleA()` flush directly to the console anyway.
